src/sst/core/baseComponent.h

```diff
@@ -404,6 +405,39 @@ class BaseComponent {
         return Factory::getFactory()->isSubComponentLoadableUsingAPI<T>(type);
     }
 
+    /**
+       Check to see if the element type loaded by the user into the.
+       specified slot is loadable with a particular API.  This will
+       only check slot index 0.  If you need to check other slots,
+       please use the SubComponentSlotInfo.
+
+       @param slot_name - Name of slot to check
+       @return True if loadable as the API specified as the template parameter
+     */
+    template<class T>
+    bool isUserSubComponentLoadableUsingAPI(const std::string& slot_name) {
+        // Get list of ComponentInfo objects and make sure that there is
+        // only one SubComponent put into this slot
+        // const std::vector<ComponentInfo>& subcomps = my_info->getSubComponents();
+        const std::map<ComponentId_t,ComponentInfo>& subcomps = my_info->getSubComponents();
+        int sub_count = 0;
+        int index = -1;
```

Comment:
Why is index needed? It doesn't look like it's used.

Reply:
The index is needed, and the 0 in the return statement has been changed to use index.  This matches the semantics of loadUserSubComponent(), which requires only one SubComponent to be loaded into the slot, but it does not require that the one SubComponent be loaded into index 0.